cub/warp/warp_exchange.cuh

```diff
@@ -36,11 +36,30 @@
 #include <cub/config.cuh>
 #include <cub/util_ptx.cuh>
 #include <cub/util_type.cuh>
+#include "specializations/warp_exchange_shfl.cuh"
+#include "specializations/warp_exchange_smem.cuh"
 
 
 CUB_NAMESPACE_BEGIN
 
 
+enum class WarpExchangeAlgorithm
+{
+  WARP_EXCHANGE_SMEM,
+  WARP_EXCHANGE_SHUFFLE,
+};
```

Comment:
I'm afraid we'll have to go with `enum WarpExchangeAlgorithm` for now. 

Current usage of other algorithms is `cub::BLOCK_LOAD_X`, having it in the enum class leads to duplication `cub::BlockLoadAlgorithm::BLOCK_LOAD_X`, which can be addressed by removing the duplication in each value: `cub::BlockLoadAlgorithm::DIRECT`. I'd prefer this, but it's not consistent with the rest of the library.